.github/workflows/basics.yml

```diff
@@ -0,0 +1,83 @@
+name: CS
+
+on:
+  # Run on all pushes and on all pull requests.
+  # Prevent the build from running when there are only irrelevant changes.
+  push:
+    paths-ignore:
+      - '**.md'
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+jobs:
+  checkcs:
+    name: 'Basic CS and QA checks'
+    runs-on: ubuntu-latest
+
+    env:
+      XMLLINT_INDENT: '	'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+          coverage: none
+          tools: cs2pr
+
+      - name: 'Composer: adjust dependencies'
+        run: |
+          # The sniff stage doesn't run the unit tests, so no need for PHPUnit.
+          composer remove --no-update --dev phpunit/phpunit --no-scripts
+          # Using PHPCS `master` as an early detection system for bugs upstream.
+          composer require --no-update --no-scripts squizlabs/php_codesniffer:"dev-master"
+          # Using WPCS `master` (=stable). This can be changed back to `dev-develop` after the WPCS 3.0.0 release.
+          composer require --no-update --no-scripts wp-coding-standards/wpcs:"dev-master"
+
+      # Validate the composer.json file.
+      # @link https://getcomposer.org/doc/03-cli.md#validate
+      - name: Validate Composer installation
+        run: composer validate --no-check-all --strict
```

Comment:
I assume the order of steps is relevant.

It seems more logical to me to validate the composer config first before making any changes to it.

_Especially with the `compose require` commands above not using the `--dev` flag._

Reply:
Good point, I will switch the order.

Re: the `--dev` flag: that flag is deprecated as `--dev` is now (and has been for a while) the default in Composer.